components/utilities/helpful.js

```diff
@@ -1,64 +1,52 @@
-import React from "react";
+import React, { useState, useEffect, useRef } from "react";
 import pull from "lodash/pull";
-
-import SuggestEdits from "./suggestEdits";
-
 import router, { withRouter } from "next/router";
 
-export default class Helpful extends React.Component {
-  constructor(props) {
-    super(props);
-    this.formRef = React.createRef();
-
-    this.handleStep = this.handleStep.bind(this);
-    this.handleOther = this.handleOther.bind(this);
-    this.submitForm = this.submitForm.bind(this);
-    this.handleRouteChange = this.handleRouteChange.bind(this);
-    this.handleImprovement = this.handleImprovement.bind(this);
-    this.handleNoteChange = this.handleNoteChange.bind(this);
-
-    this.state = {
-      step: 0,
-      other: false,
-      helpful: true,
-      improvements: [],
-      notes: "",
-      improvementsString: "",
-      moreExamples: false,
-      clearerSteps: false,
-      moreInformation: false,
-      other: false,
-    };
-  }
+import SuggestEdits from "./suggestEdits";
 
-  handleStep(newStep) {
-    this.setState({ step: newStep });
+const Helpful = ({ slug, sourcefile }) => {
+  const formRef = useRef();
+  const [state, setState] = useState({
+    step: 0,
+    other: false,
+    helpful: true,
+    improvements: [],
+    notes: "",
+    improvementsString: "",
+    moreExamples: false,
+    clearerSteps: false,
+    moreInformation: false,
+    other: false,
+  });
```

Comment:
Break into multiple states. (Although [it's ok to group values that change together](https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables))

Reply:
Makes sense! Fixed on 18e3592, and kept the feedback values in one single state update per the link you recommended, LMKWYT!